### ISSUE TITLE
[Bug]: CREATE INDEX ONLY ON hypertable creates index on chunks

### ISSUE THREAD

**Issue by pmwkaa** [timescale/timescaledb#5908]
### What type of bug is this?

Incorrect result

### What subsystems and features are affected?

Command processing

### What happened?

Currently there is no way to create index only on hypertable main table excluding chunks. Essentially `CREATE INDEX ONLY ON` clause is not supported and being ignored.

### TimescaleDB version affected

2.11.1

### PostgreSQL version used

15.2

### What operating system did you use?

Debian

### What installation method did you use?

Source

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
SQL
tsdb=> CREATE INDEX _hyper_23_5068_chunk_best_idx ON _timescaledb_internal._hyper_23_5068_chunk(val);
CREATE INDEX
tsdb=> \d _timescaledb_internal._hyper_23_5068_chunk
         Table "_timescaledb_internal._hyper_23_5068_chunk"
 Column |           Type           | Collation | Nullable | Default
--------+--------------------------+-----------+----------+---------
 time   | timestamp with time zone |           | not null |
 val    | integer                  |           |          |
 foo    | integer                  |           |          |
 bar    | integer                  |           |          |
Indexes:
    "_hyper_23_5068_chunk_best_idx" btree (val)
    "_hyper_23_5068_chunk_index_name_test_time_idx" btree ("time" DESC)
Check constraints:
    "constraint_2779" CHECK ("time" >= '2022-12-24 00:00:00+00'::timestamp with time zone AND "time" < '2023-01-23 00:00:00+00'::timestamp with time zone)
Inherits: index_name_test
```

``` SQL
tsdb=> CREATE INDEX best_idx ON index_name_test(val);
CREATE INDEX
tsdb=> \d _timescaledb_internal._hyper_23_5068_chunk
         Table "_timescaledb_internal._hyper_23_5068_chunk"
 Column |           Type           | Collation | Nullable | Default
--------+--------------------------+-----------+----------+---------
 time   | timestamp with time zone |           | not null |
 val    | integer                  |           |          |
 foo    | integer                  |           |          |
 bar    | integer                  |           |          |
Indexes:
    "_hyper_23_5068_chunk_best_idx" btree (val)
    "_hyper_23_5068_chunk_best_idx_1" btree (val)
    "_hyper_23_5068_chunk_index_name_test_time_idx" btree ("time" DESC)
Check constraints:
    "constraint_2779" CHECK ("time" >= '2022-12-24 00:00:00+00'::timestamp with time zone AND "time" < '2023-01-23 00:00:00+00'::timestamp with time zone)
Inherits: index_name_test
```
```

**Comment by jflambert** [comment]
@pmwkaa please, can you help me understand if this fix is good for me

does this basically create an index for future chunks only?

Is this in 2.11.2?

Context: I'm trying to index some hypertables, and it's taking forever. I'd be happy with only indexing future chunks, since retention policies eventually cleanup the past.

**Comment by jflambert** [comment]
To partially answer my question, this fix is in 2.12.0, not 2.11.2

Would still like clarity about the first question.

**Comment by fabriziomello** [comment]
> To partially answer my question, this fix is in 2.12.0, not 2.11.2
> 
> 
> 
> Would still like clarity about the first question.

This is part of 2.11.2.
